source/Interfaces/MainInterface.cpp

```diff
@@ -0,0 +1,297 @@
+                switch (symbol) {
+                    case '#':
+                        cellRect.setTexture(&wallTexture);
+                        if (isVerticalWall) {
+//                                sf::Transform transform;
+////                                transform.rotate(270.0f, sf::Vector2f(cellPosX + cellSize / 2.0f, cellPosY + cellSize / 2.0f));
+//
+//                                transform.rotate(270.0f,sf::Vector2f(cellPosX + cellSize / 2.0f, cellPosY + cellSize / 2.0f));
+//                                mWindow.draw(cellRect, transform);
+                            mWindow.draw(cellRect);
+
+                        } else {
+                            mWindow.draw(cellRect);
+                        }
+                        break;
+
+                    case ' ':
+                        cellRect.setFillColor(sf::Color::Black);
+                        mWindow.draw(cellRect);
+                        break;
+
+                    case '*':
+                        cellRect.setTexture(&troll);
+                        cellRect.setFillColor(sf::Color::Cyan);
+                        cell.setFillColor(sf::Color::Black);
+                        mWindow.draw(cell);
+                        mWindow.draw(cellRect);
+                        break;
+
+                    case '@':
+                        cellRect.setTexture(&troll);
+                        cellRect.setFillColor(sf::Color::Red);
+                        cell.setFillColor(sf::Color::Black);
+                        mWindow.draw(cell);
+                        mWindow.draw(cellRect);
+                        break;
+
+                }
+            }
+
+        }
+        // Display everything
+        menuBtn.drawTo(mWindow);
+        inventoryBTN.drawTo(mWindow);
+        exitBtn.drawTo(mWindow);
+        mWindow.display();
+    }
+
+    /**
+    * @brief Updates the graphical representation of the maze grid.
+    *
+    * @param grid         The WorldGrid representing the maze.
+    * @param type_options The type options for symbols.
+    * @param item_set     The set of items in the maze.
+    * @param agent_set    The set of agents in the maze.
+    */
+    void MainInterface::UpdateGrid(const WorldGrid &grid,
+                                   const type_options_t &type_options,
+                                   const item_set_t &item_set,
+                                   const agent_set_t &agent_set) {
+        DrawGrid(grid, type_options, item_set, agent_set);
+    }
+
+    /**
+    * @brief Handles user input for selecting actions.
+    *
+    * @param grid         The WorldGrid representing the maze.
+    * @param type_options The type options for symbols.
+    * @param item_set     The set of items in the maze.
+    * @param agent_set    The set of agents in the maze.
+    */
+    size_t MainInterface::SelectAction(const WorldGrid &grid,
```

Comment:
This function also needs to be shortened by using helpers.  It should be easy by adding functions like `DoKeypress`, `DoResize`, `DoMouseMove`, etc.